src/App.tsx

```diff
@@ -1,24 +1,137 @@
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const filteredList = useMemo(() => {
+    return todoList && todoList.filter((todo) => {
+      switch (selectedFilter) {
+        case Select.ACTIVE:
+          return !todo.completed;
+        case Select.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [selectedFilter, todoList]);
+
+  const countItemLeft = todoList?.filter(todo => !todo.completed).length;
+
+  const deleteClickHandlerItem = (
+    id: number,
+    loaderState: (loaderTodo: boolean) => void,
+  ) => {
+    loaderState(true);
+
+    if (id) {
+      postDelete(id)
+        .then(() => {
+          const newArray = todoList ? [...todoList] : [];
+          const objFindIndex = newArray.findIndex(obj => obj.id === id);
+
+          newArray.splice(objFindIndex, 1);
+          setTodoList(newArray);
+          loaderState(false);
+        })
+        .catch(() => {
+          setTypeError(Errors.REMOVE);
+        });
+    }
+  };
+
+  const deleteClickHandlerFooter = () => {
+    const completedTodo
+    = todoList?.filter(todo => todo.completed).map(todo => todo.id);
+
+    if (completedTodo) {
+      setLoadingTodoIds(completedTodo);
+    }
+
+    completedTodo?.forEach(todo => {
+      postDelete(todo)
+        .then(() => {
+          if (todoList) {
+            setTodoList(
+              todoList.filter(todoFilter => {
+                return !completedTodo.includes(todoFilter.id);
+              }),
+            );
+          }
+        })
+        .catch(() => {
+          setTypeError(Errors.REMOVE);
+        });
+    });
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodoList)
+      .catch(() => {
+        setTypeError(Errors.UPDATE);
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          USER_ID={USER_ID}
+          counterItemLeft={countItemLeft}
+          setTypeError={setTypeError}
+          todoList={todoList}
+          setTempTodo={setTempTodo}
+          setTodoList={setTodoList}
+        />
+
+        {todoList && (
+          <TodoList
+            filteringList={filteredList}
+            tempTodo={tempTodo}
+            loadingTodoIds={loadingTodoIds}
+            deleteClickHandler={deleteClickHandlerItem}
+          />
+        )}
+
+        {(todoList?.length !== 0 && todoList) && (
```

Comment:
It does not work because 0 remains
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/68865653/d43aeb9b-36cd-4856-a588-a0b37c91627c)